src/app/WriteClient.cpp

```diff
@@ -350,7 +359,9 @@ CHIP_ERROR WriteClientHandle::SendWriteRequest(SessionHandle session, System::Cl
 {
     CHIP_ERROR err = mpWriteClient->SendWriteRequest(session, timeout);
 
-    if (err == CHIP_NO_ERROR)
+    // The InteractionModelEngine will only close the writeClienHandle upon reception of a writeResponse
```

Comment:
What we're really doing here is transferring ownership of the underlying `WriteClient` to the IM layer and for the IM to subsequently manage its lifetime. For groupcast writes, there is no transfer of ownership since the interaction is done upon transmission of the action.

Could we please update the comments to that effect?

Reply:
Done